src/electionguard_gui/components/key_ceremony_details_component.py

```diff
@@ -47,11 +46,98 @@ def can_join_key_ceremony(self, key_ceremony) -> bool:
 
     def watch_key_ceremony(self, key_ceremony_id: str) -> None:
         db = self.db_service.get_db()
+        # retrieve and send the key ceremony to the client
+        self.on_key_ceremony_changed(key_ceremony_id)
         self.log.debug(f"watching key ceremony '{key_ceremony_id}'")
+        # start watching for key ceremony changes from guardians
         self._key_ceremony_service.watch_key_ceremonies(
-            db, key_ceremony_id, lambda: self.refresh_ceremony(db, key_ceremony_id)
+            db, key_ceremony_id, self.on_key_ceremony_changed
         )
 
+    def on_key_ceremony_changed(self, key_ceremony_id: str) -> None:
```

Comment:
I really dislike how big this file is getting.  I will be refactoring as I go, but I'm still ruminating on the best way to go about it.  Thoughts are welcome, else I'll figure it out as it continues to take shape.